### PR TITLE
Support for a group help button and callback

### PR DESCRIPTION
**Pull request by samelhusseini**
This PR adds support for a group help button that calls a callback when a user clicks on it. 

It also adds support for a ``workspaceevent`` which currently only passed up the chain in an iframeworkspace. 
The two events supported: 
- pxt.events.UIEvent for the groupHelpCallback
- pxt.events.CreateEvent for when blocks are created. 

PR also pulls in a Blockly build for support for the help icon with PRs: 
- https://github.com/Microsoft/pxt-blockly/pull/163
- https://github.com/Microsoft/pxt-blockly/pull/166

https://github.com/playi/pxt-wonder/issues/338 and https://github.com/playi/pxt-wonder/issues/337

### CODE REVIEW

File: pxteditor/events.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+namespace pxt.events {
+
+    export type EditorType = 'blocks' | 'js';
```
Comment (pelikhan) [inline pxteditor/events.ts]: to stay consistent, use file extension, ``.ts``

File: pxteditor/events.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    export interface Event {
+        type: string;
+        editor: EditorType;
+    }
+
+    export abstract class AbstractEvent implements Event {
```
Comment (pelikhan) [inline pxteditor/events.ts]: do we need classes, we can typically define these messages as bare interface

File: pxtlib/service.ts [hunk 3]
```diff
@@ -845,6 +846,14 @@ namespace ts.pxtc {
                 res.groupIcons = undefined;
             }
         }
+        if (res.groupHelp) {
+            try {
+                res.groupHelp = JSON.parse(res.groupHelp as any);
+            }
+            catch (e) {
+                res.groupHelp = undefined;
```
Comment (pelikhan) [inline pxtlib/service.ts]: report issue

File: webapp/src/blocks.tsx [hunk 4]
```diff
@@ -378,6 +379,7 @@ export class Editor extends toolboxeditor.ToolboxEditor {
                 if (ev.xml.tagName == 'SHADOW')
                     this.cleanUpShadowBlocks();
                 this.parent.setState({ hideEditorFloats: false });
+                workspace.fireEvent(new pxt.events.CreateEvent('blocks', blockId));
```
Comment (pelikhan) [inline webapp/src/blocks.tsx]: fireEvent is optional?

File: pxteditor/events.ts (hunk 1 above)
Comment (samelhusseini) [reply pxteditor/events.ts]: Sounds good.

File: pxteditor/events.ts (hunk 2 above)
Comment (samelhusseini) [reply pxteditor/events.ts]: Switching to interfaces

File: webapp/src/blocks.tsx (hunk 4 above)
Comment (samelhusseini) [reply webapp/src/blocks.tsx]: Optional in impl not in workspace

File: pxteditor/events.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+namespace pxt.events {
```
Comment (pelikhan) [inline pxteditor/events.ts]: can you put this under ``pxt.editor.events`` ? I think all editor stuff is there.

Comment (samelhusseini) [reply pxteditor/events.ts]: sure.

File: pxteditor/events.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+    export type EditorType = 'blocks' | 'ts';
+
+    export interface PXTEvent {
```
Comment (pelikhan) [inline pxteditor/events.ts]: don't use the product name here. we're already under the pxt namespace. ``HostEvent``?

Comment (samelhusseini) [reply pxteditor/events.ts]: Sorted.

**Pull request merged**